source/tutorials/nix-language.md

```diff
@@ -0,0 +1,325 @@
+# Reading the Nix language
+
+The Nix language is used to declare packages and configurations for the Nix package manager.
+
+You will quickly encounter Nix language expressions that may look very complicated.
+Yet, the language has only few basic concepts that can be combined arbitrarily.
+Use this guide to learn reading code in Nix language and understand its structure.
+
+Imagine the Nix language as JSON with functions.
+Its purpose is to define structured data, and functions help with conveniently producing more complex data.
+All of the language constructs can be composed arbitrarily.
+
+### Attribute sets
+
+The primary data structure is the attribute set, a collection of key-value pairs.
+
+JSON:
+
+    {
+      "string": "hello",
+      "integer": 1,
+      "float": 3.141,
+      "bool": true,
+      "null": null,
+      "list": [1, 2, 3],
+      "set": {
+        "a": "hello",
+        "b": 1,
+        "c": 2.718,
+        "d": false
+      }
+    }
+
+Nix language:
+
+    {
+      string = "hello";
+      integer = 1;
+      float = 3.141;
+      bool = true;
+      null = null;
+      list = [ 1 2 3 ];
+      attribute-set = {
+        a = "hello";
+        b = 2;
+        c = 2.718;
+        d = false;
+      }; # comments are supported
+    }
+
+Note: List elements are separated by white space.
+
+Note: The outermost expression does not and must not have a semicolon.
+
+See attribute [naming rules]().
+
+Nix has additional primitives:
+
+### String
+
+#### Multiline string
+
+    nix-repl> ''
+              string
+              spanning
+              multiple
+              lines
+              ''
+    "string\nspanning\nmultiple\nlines\n"
+
+Whitespace that prefixes all lines is stripped automatically.
+
+Multiline strings ease the use of quotes and special characters.
+
+    nix-repl> ''
+              "double quoted" and 'single quoted'
+              ''
+    "\"double quoted\" and 'single quoted'\n"
+
+#### String interpolation
+
+    nix-repl> let
+              name = "Nix";
+              in
+              "hello ${name}"
+    "hello Nix"
+
+See escaping rules.
+
+### Path
+
+#### Relative path
+
+    ./relative/path
+
+The path is relative to the file containing the expression.
+
+#### Absolute path
+
+    /absolute/path
+
+#### Home directory path
+
+    ~/path/in/home/directory
+
+#### Named paths
+
+Also known as “angle bracket syntax”.
+
+    <name>
+
+Examples:
+
+    # write a Nix expression to a file
+    echo 123 > nix_path_file
+
+    # set `NIX_PATH` to point a name to that file
+    env NIX_PATH=name_in_nix_path=nix_path_file nix repl
+
+    nix repl> import <nix_path_file>
+    123
+
+Nix simply reads and parses the `NIX_PATH` environment variable.
+Entries are separated by colons (`:`).
+
+Note: The value of a named path depends on external system state. Therefore it is recommended not to use them.
```

Comment:
As a new nix user, I find comments like this a bit scary because I see examples (especially ones involving `<nixpkgs>`) all over the place. Are these examples doing it wrong? Or should the "recommended not to use them" advice be qualified?

Reply:
> Are these examples doing it wrong?

IMHO: Yes. At best it's a sloppy short-hand. See the [pinning tutorial](https://github.com/NixOS/nix.dev/blob/21ed44040211f92edc179f1017ab96a74a4ebdd7/source/tutorials/towards-reproducibility-pinning-nixpkgs.md?plain=1#L5-L18) for reasoning.

```suggestion
Note:
The value of a named path depends on external system state.
That means, the same Nix expression can produce different results, which may be very confusing.
Therefore {ref}`we recommend to avoid named paths<Towards reproducibility: pinning Nixpkgs>`.
```